libs/core-ui/src/lib/util/calculateAUCData.ts

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { localization } from "@responsible-ai/localization";
+import { SeriesOptionsType } from "highcharts";
+import { orderBy, unzip } from "lodash";
+
+import { IDataset } from "../Interfaces/IDataset";
+
+interface IPoint {
+  x: number;
+  y: number;
+}
+export interface IROCData {
+  points: IPoint[];
+}
+
+function getStaticROCData(): SeriesOptionsType[] {
+  return [
+    {
+      data: [
+        { x: 0, y: 0 },
+        { x: 0, y: 1 },
+        { x: 1, y: 1 }
+      ],
+      name: localization.Interpret.Charts.Ideal,
+      type: "line"
+    },
+    {
+      data: [
+        { x: 0, y: 0 },
+        { x: 1, y: 1 }
+      ],
+      name: localization.Interpret.Charts.Random,
+      type: "line"
+    }
+  ];
+}
+
+export function calculatePerClassROCData(
+  probabilityY: number[],
+  binY: number[]
+): IROCData {
+  const thresholds = probabilityY.sort();
+  const points: IPoint[] = [];
+  for (const threshold of thresholds) {
```

Comment:
100*n effort, still linear

My suggestion would be to sort (nlogn) and then step through (n) and build the entire curve in one walkthrough because with ordering you know exactly how it changes the percentage when increasing thresholds 🙂 This still needs the convex hull cleanup, of course, because some points are below others and irrelevant.

I suspect since our n is quite small the log n factor won't really hurt.

The convex hull cleanup would have more points, though (about n instead of 101), so that could take longer.